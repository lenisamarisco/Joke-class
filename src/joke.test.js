import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import Joke from "./joke.js";

describe("Joke Component", () => {
  test("renders correctly with given props", () => {
    const { container } = render(
      <Joke vote={() => {}} votes={5} text="Funny joke text" id={1} />
    );
    expect(container).toMatchSnapshot();
  });

  test("displays joke text and votes correctly", () => {
    render(<Joke vote={() => {}} votes={10} text="Hilarious joke" id={2} />);
    expect(screen.getByText("Hilarious joke")).toBeInTheDocument();
    expect(screen.getByText("10")).toBeInTheDocument();
  });

  test("calls vote function with correct arguments when upvote and downvote buttons are clicked", () => {
    const mockVote = jest.fn();
    render(<Joke vote={mockVote} votes={0} text="Vote testing joke" id={3} />);

    // Upvote button
    fireEvent.click(screen.getByRole("button", { name: /thumbs-up/i }));
    expect(mockVote).toHaveBeenCalledWith(3, 1);

    // Downvote button
    fireEvent.click(screen.getByRole("button", { name: /thumbs-down/i }));
    expect(mockVote).toHaveBeenCalledWith(3, -1);
  });
});
