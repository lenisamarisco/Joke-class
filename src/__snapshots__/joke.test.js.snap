// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Joke Component renders correctly with given props 1`] = `
<div>
  <div
    class="Joke"
  >
    <div
      class="Joke-votearea"
    >
      <button>
        <i
          class="fas fa-thumbs-up"
        />
      </button>
      <button>
        <i
          class="fas fa-thumbs-down"
        />
      </button>
      5
    </div>
    <div
      class="Joke-text"
    >
      Funny joke text
    </div>
  </div>
</div>
`;
